packages/common/src/index-api.ts

```diff
@@ -27,32 +27,54 @@ export type Pagination = ForwardPagination | BackwardPagination
 /**
  * Boolean field value filter
  */
-export type BooleanValueFilter = { isNull: boolean } | { equalTo: boolean }
-
-/**
- * Enum field value filter
- */
-export type EnumValueFilter =
+export type BooleanValueFilter =
   | { isNull: boolean }
-  | { equalTo: string }
-  | { notEqualTo: string }
-  | { in: Array<string> }
-  | { notIn: Array<string> }
+  | { equalTo: boolean }
+  | { notEqualTo: boolean }
 
 /**
- * String or number field value filter
+ * Common type for multiple value filters
  */
-export type ScalarValueFilter<T extends string | number = string | number> =
+export type CommonValueFilter<T> =
   | { isNull: boolean }
   | { equalTo: T }
   | { notEqualTo: T }
   | { in: Array<T> }
   | { notIn: Array<T> }
-  | { lessThan: T }
-  | { lessThanOrEqualTo: T }
+
+/**
+ * Enum field value filter
+ */
+export type EnumValueFilter = CommonValueFilter<string>
+
+/**
+ * Supported greater than value conditions
+ */
+export type GreaterThanValueFilter<T extends string | number> =
   | { greaterThan: T }
   | { greaterThanOrEqualTo: T }
 
+/**
+ * Supported less than value conditions
+ */
+export type LessThanValueFilter<T extends string | number> =
+  | { lessThan: T }
+  | { lessThanOrEqualTo: T }
+
+/**
+ * Range value filter using a required greater than or less than value filter
+ * and optional other boundary
+ */
+export type RangeValueFilter<T extends string | number> =
+  | (GreaterThanValueFilter<T> & Partial<LessThanValueFilter<T>>)
```

Comment:
why does the RHS need to be a `Partial`?  Does `Partial` even do anything on a type with only one field?

Reply:
`Partial` makes all the fields of the object optional, so here the type ensures that when setting a range boundary, only another valid boundary can be set, but it's optional.